### PR TITLE
Migrate Starknet Account Preset tests

### PR DESCRIPTION
**Pull request by immrsd**
- Fixes [#1006](https://github.com/OpenZeppelin/cairo-contracts/issues/1006)
- Makes the `assert_entrypoint_not_found_error` function work with `SyscallResult`

### CODE REVIEW

File: src/tests/utils/common.cairo [hunk 1]
```diff
@@ -38,13 +38,21 @@ pub impl IntoBase16String<T, +Into<T, felt252>> of IntoBase16StringTrait<T> {
 
 /// Asserts that the panic data is an "Entrypoint not found" error, following the starknet foundry
 /// emitted error format.
-pub fn assert_entrypoint_not_found_error(
-    panic_data: Array<felt252>, selector: felt252, contract_address: ContractAddress
+pub fn assert_entrypoint_not_found_error<T, +Drop<T>>(
```
Comment (ericnordelo) [inline src/tests/utils/common.cairo:41]: Good call.

File: src/tests/utils/common.cairo [hunk 2]
```diff
@@ -38,13 +38,21 @@ pub impl IntoBase16String<T, +Into<T, felt252>> of IntoBase16StringTrait<T> {
 
 /// Asserts that the panic data is an "Entrypoint not found" error, following the starknet foundry
 /// emitted error format.
```
Comment (ericnordelo) [inline src/tests/utils/common.cairo:40]: We need to update this comment.

File: src/tests/presets/test_account.cairo [hunk 3]
```diff
@@ -426,113 +477,97 @@ fn test_multicall() {
...
+#[feature("safe_dispatcher")]
 fn test_v2_missing_camel_selector() {
-    let v1 = setup_dispatcher();
-    let v2_class_hash = V2_CLASS_HASH();
+    let key_pair = KEY_PAIR();
+    let (account_address, v1_dispatcher) = setup_dispatcher(key_pair);
+
+    let v2_class_hash = declare_v2_class();
+    start_cheat_caller_address(account_address, account_address);
+    // set_contract_and_caller(v1.contract_address);
```
Comment (andrew-fleming) [inline src/tests/presets/test_account.cairo:543]:
```suggestion
```

File: src/tests/utils/common.cairo [hunk 4]
```diff
@@ -36,15 +36,23 @@ pub impl IntoBase16String<T, +Into<T, felt252>> of IntoBase16StringTrait<T> {
     }
 }
 
-/// Asserts that the panic data is an "Entrypoint not found" error, following the starknet foundry
-/// emitted error format.
-pub fn assert_entrypoint_not_found_error(
-    panic_data: Array<felt252>, selector: felt252, contract_address: ContractAddress
+/// Asserts that the syscall result of a call failed with an "Entrypoint not found" error,
```
Comment (andrew-fleming) [inline src/tests/utils/common.cairo:39]:
```suggestion
/// Asserts that the syscall result of a call failed with an "Entrypoint not found" error,
```

**Review by andrew-fleming — APPROVED** [review]
LGTM! I left two tiny suggestions

**Pull request merged**